### ISSUE TITLE
Losing nested embedded document short name in ClassMetadataInfo reflFields

### ISSUE THREAD

**Issue by pilou** [doctrine/mongodb-odm#1428]
We have nested embedded documents like this
```
/** @ODM\Document */
class Document
{
    /** @ODM\Id */
    public $id;

    /** @ODM\EmbedOne(targetDocument="EmbeddedDocument") */
    public $embedded;
}

/** @ODM\EmbeddedDocument */
class EmbeddedDocument
{
    /** @ODM\EmbedOne(targetDocument="NestedEmbeddedDocument", name="shortNameThatDoesntExist") */
    public $nestedEmbedded;
}

/** @ODM\EmbeddedDocument */
class NestedEmbeddedDocument
{
}
```
And we flush,
then replace the middle embed instance (`EmbeddedDocument`), having the same instance `NestedEmbeddedDocument`) set to it
 then flush.
Like this:
...
        $owner = new Document();
        $nestedEmbedded = new NestedEmbeddedDocument();
        $this->dm->persist($owner);
        $this->dm->flush();

        $owner->embedded = new EmbeddedDocument();
        $owner->embedded->nestedEmbedded = $nestedEmbedded;
        $this->dm->flush();

        $owner->embedded = new EmbeddedDocument();
        $owner->embedded->nestedEmbedded = $nestedEmbedded;
        $this->dm->flush();
...
This will breake ClassMetaInfo->setFieldValue as the `shortNameThatDoesntExist` wont exits in `reflFields`.

Breaking test case coming soon...

**Comment by pilou** [comment]
Occurs on php 5.6.18 and 7.0.0 too

**Comment by alcaeus** [comment]
This has been fixed in #1432. Thanks!
